jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/handler/SendSignalAction.java

```diff
@@ -0,0 +1,81 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.bpmn2.handler;
+
+import org.drools.core.process.instance.WorkItemManager;
+import org.drools.core.process.instance.impl.WorkItemImpl;
+import org.jbpm.process.core.event.EventTransformerImpl;
+import org.jbpm.process.instance.InternalProcessRuntime;
+import org.jbpm.process.instance.impl.Action;
+import org.jbpm.process.instance.impl.util.VariableUtil;
+import org.jbpm.workflow.core.impl.NodeImpl;
+import org.jbpm.workflow.core.node.Transformation;
+import org.kie.api.runtime.manager.RuntimeManager;
+import org.kie.api.runtime.process.ProcessContext;
+
+public class SendSignalAction implements Action {
+
+    private String signalName;
+    private String varName;
+    private boolean isAsync;
+    private NodeImpl node;
+
+    public SendSignalAction(NodeImpl node, String variable, String signalName, boolean isAsync) {
+        this.node = node;
+        this.varName = variable;
+        this.signalName = signalName;
+        this.isAsync = isAsync;
+    }
+
+    @Override
+    public void execute(ProcessContext kcontext) throws Exception {
+        Object tVariable = kcontext.getVariable(varName);
+        Transformation transformation = (Transformation) kcontext.getNodeInstance().getNode()
+                .getMetaData().get("Transformation");
+        if (transformation != null) {
+            tVariable = new EventTransformerImpl(transformation).transformEvent(tVariable);
+        }
+        String signalType = VariableUtil.resolveVariable(isAsync ? "ASYNC-" + signalName : signalName, kcontext
+                .getNodeInstance());
+        String scope = (String) node.getMetaData("customScope");
+        if ("processInstance".equalsIgnoreCase(scope)) {
+            kcontext.getProcessInstance().signalEvent(signalType, tVariable);
+        } else if ("runtimeManager".equalsIgnoreCase(scope) || "project".equalsIgnoreCase(scope)) {
+            ((RuntimeManager) kcontext.getKieRuntime().getEnvironment().get("RuntimeManager")).signalEvent(signalType,
+                    tVariable);
+        } else if ("external".equalsIgnoreCase(scope)) {
+            ((InternalProcessRuntime) kcontext.getKieRuntime()).getProcessEventSupport().fireOnSignal(kcontext
+                    .getProcessInstance(), kcontext.getNodeInstance(), kcontext.getKieRuntime(), signalName, tVariable);
+            WorkItemImpl workItem = new WorkItemImpl();
+            workItem.setName("External Send Task");
```

Comment:
Signal use External Send Task

Reply:
the work item handler is not required. remove it.